lib/geo.js

```diff
@@ -10,23 +10,31 @@ var assert = require('assert');
  */
 
 exports.nearFilter = function nearFilter(where) {
+  var keyPath = [];
   var result = false;
 
-  if (where && typeof where === 'object') {
-    Object.keys(where).forEach(function(key) {
-      var ex = where[key];
-
-      if (ex && ex.near) {
-        result = {
-          near: ex.near,
-          maxDistance: ex.maxDistance,
-          unit: ex.unit,
-          key: key,
-        };
-      }
-    });
+  searchForNear(where);
+
+  function searchForNear(node) {
+    if (node && typeof node === 'object') {
+      Object.keys(node).forEach(function(key) {
+        var ex = node[key];
+        keyPath.push(key);
+
+        if (ex && ex.near) {
+          result = {
+            near: ex.near,
+            maxDistance: ex.maxDistance,
+            unit: ex.unit,
+            key: keyPath.slice(0),
```

Comment:
it normally returns a string that indicates the property on the where clause that the `buildNearFilter` of the connector uses to mutate `near` (see MongoDB connector PR): https://github.com/strongloop/loopback-connector-mongodb/pull/273

The current implementation of nearFilter assumes the `where` filter is flat... so it doesn't support geo searches with `and` logic. In order to support `and`... the `nearFilter` needs to do a deep search to find the nested `near` property... thus the array so that `buildNearFilter` can find the nested `near` property.